conda_build/build.py

```diff
@@ -1276,7 +1276,8 @@ def test(recipedir_or_package_or_metadata, config, move_broken=True):
 
     warn_on_use_of_SRC_DIR(metadata)
 
-    config.compute_build_id(metadata.name())
+    if config.set_build_id:
```

Comment:
Also I'd like to add that in the example showed on https://github.com/conda/conda-build/issues/2101 if I replace the source url with a local path the work dir is overwritten and the example works